bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/util/ContextRequest.java

```diff
@@ -145,4 +146,34 @@ public static boolean isMessageType(final WsMessageContext ctx, final String typ
       }
       return Optional.empty();
    }
+
+   public static Optional<Message<String>> readMessage(final WsMessageContext ctx) {
+      try {
+         JsonNode json = JavalinJackson.getObjectMapper().readTree(ctx.message());
+         if (!json.has(JsonResponseMember.TYPE)) {
+            error(ctx, "Missing message type");
+            return Optional.empty();
+         }
+
+         JsonNode jsonTypeNode = json.get(JsonResponseMember.TYPE);
+         String jsonType = !jsonTypeNode.asText().isEmpty() ? jsonTypeNode.asText() : jsonTypeNode.toString();
+         if (jsonType.equals("{}")) {
```

Comment:
wasn't there a json util which checked for empty string?

Reply:
`JsonNode` offers an `isEmpty` method itself afaik.
```suggestion
         if (jsonType.isEmpty()) {
```

Reply:
The pattern seems similar to what already existed in this class. As far as I understand, JsonNode.isEmpty() is not strictly equivalent, because it would match both `[]` and `{}`, as well as `""`.

But I think 'type' is expected to be a string anyway, so anything other than non-empty text should be invalid. If I read the code correctly, we currently accept e.g. `"type": {"a": true, "b": "value"}` as a valid type (Parsed a a string: `{"a": true, "b": "value"}`), which doesn't make a lot of sense.

Using `jsonTypeNode.isTextual() && !jsonTypeNode.isEmpty()` would IMO be more consistent.

@cdamus WDYT?

Reply:
Yes, I prefer Camille's last suggestion as it's more consistent with the abstract meaning. WBN not to compare things to the concrete JSON syntax.